operator/cmd/mesh/install.go

```diff
@@ -146,7 +157,10 @@ func runApplyCmd(cmd *cobra.Command, rootArgs *rootArgs, iArgs *installArgs, log
 	if err != nil {
 		return err
 	}
-	setFlags := applyFlagAliases(iArgs.set, iArgs.manifestsPath, iArgs.revision)
+	if iArgs.revision != "" && !existingDefaultRevision(clientset, iArgs.istioNamespace) {
+		iArgs.defaultRevision = true
```

Comment:
nice! one concern though is I don't think this would work with external istiod. For example, I am deploying just remote chart and have `injectionURL` set. The Service will never exist, so now every install will set it as default